proto/keyserver/v1/keyserver.proto

```diff
@@ -0,0 +1,43 @@
+syntax = "proto3";
+
+package keyserver;
+
+// AccountKeyService is a key service for storing cosmos account keys
+// these keys are assumed to be `secp256k1` keys
+service AccountKeyService {
+    rpc NewKey(NewKeyRequest) returns (NewKeyResponse);
+    rpc Sign(SignRequest) returns (SignResponse);
+    rpc PubKey(PubKeyRequest) returns (PubKeyResponse);
+}
+
+message NewKeyRequest {
+    string name = 1;
+    string derivation_path = 2;
+}
+
+message NewKeyResponse {
+    bytes pub_key = 1;
+    string error = 2;
+}
+
+message SignRequest {
```

Comment:
is account number the key identifier here? 

Im guessing this would work like, lens queries the chain for the account number then the message to sign is sent to the key signer?

Reply:
Yeah, lens would query the chain and get this info. I expect the account number for same key pair to be different in different chains, that's why it's sent on the SignRequest and not in KeyInfo.